assets/marketplace/oss-existingvpc.yaml

```diff
@@ -0,0 +1,1064 @@
+        Count: 2
+        Timeout: PT10M
+    UpdatePolicy:
+      AutoScalingRollingUpdate:
+        MaxBatchSize: 2
+        MinInstancesInService: 0
+        MinSuccessfulInstancesPercent: 100
+        PauseTime: PT5M
+        WaitOnResourceSignals: true
+    Properties:
+      # Use IDs of the subnets created in two AZs
+      # to let AWS know we want instances to be placed
+      # in those subnets.
+      VPCZoneIdentifier:
+        - {Ref: AuthSubnetA}
+        - {Ref: AuthSubnetB}
+      LaunchConfigurationName: {Ref: AuthLaunchConfig}
+      MinSize: 2
+      MaxSize: 2
+      TargetGroupARNs:
+        - {Ref: AuthTargetGroup}
+
+  #
+  # Proxy Setup
+  #
+  ProxySecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      VpcId: {Ref: VPC}
+      GroupDescription: Allow proxy and auth server access
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          Description: SSH emergency access
+          FromPort: 22
+          ToPort: 22
+          CidrIp: 0.0.0.0/0
+
+        - IpProtocol: tcp
+          Description: Port for SSH clients
+          FromPort: 3023
+          ToPort: 3023
+          CidrIp: 0.0.0.0/0
+
+        - IpProtocol: tcp
+          Description: Port for remote trusted clusters
+          FromPort: 3024
+          ToPort: 3024
+          CidrIp: 0.0.0.0/0
+
+        - IpProtocol: tcp
+          Description: Port for HTTPS connections
+          FromPort: 3080
+          ToPort: 3080
+          CidrIp: 0.0.0.0/0
+
+      SecurityGroupEgress:
+        - IpProtocol: -1
+          Description: Egress is allowed anywhere
+          FromPort: 0
+          ToPort: 0
+          CidrIp: 0.0.0.0/0
+
+  ProxyLB:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      Type: network
+      LoadBalancerAttributes:
+        - Key: load_balancing.cross_zone.enabled
+          Value: 'true'        
+      Subnets:
+        - Ref: ProxySubnetA
+        - Ref: ProxySubnetB
+
+  # Create A DNS entry pointing to load balancer
+  ProxyDNS:
+    Type: AWS::Route53::RecordSetGroup
+    Properties:
+      HostedZoneId: !Ref HostedZoneID
+      Comment: Zone Alias for DNS
+      RecordSets:
+      - Name: !Ref DomainName
+        Type: A
+        AliasTarget:
+          HostedZoneId: !GetAtt ProxyLB.CanonicalHostedZoneID
+          DNSName: !GetAtt ProxyLB.DNSName
+
+  ProxyLBListener:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    Properties:
+      DefaultActions:
+      - Type: forward
+        TargetGroupArn:
+          Ref: ProxyWebTargetGroup
+      LoadBalancerArn:
+        Ref: ProxyLB
+      Port: '443'
+      Protocol: TCP
+
+  ProxyWebTargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties:
+      HealthCheckIntervalSeconds: 30
+      HealthCheckProtocol: TCP
+      HealthCheckTimeoutSeconds: 10
+      HealthyThresholdCount: 4
+      UnhealthyThresholdCount: 4
+      Port: 3080
+      Protocol: TCP
+      VpcId: {Ref: VPC}
+
+  ProxyLaunchConfig: 
```

Comment:
would be cool if we found a way to re-use ASG configs, otherwise we will end up with two copies of the configuration

Reply:
I think the problem here is the restriction of us only being able to use one big YAML file for the marketplace - so no includes. We could potentially put the ProxyLaunchConfig stuff in a separate file, then write some kind of script to substitute a placeholder value in each YAML file with the included config?